### ISSUE TITLE
New xarray and mypy issues

### ISSUE THREAD

**Issue by gareth-j** [openghg/openghg#374]
The latest version of xarray (2022.6.0) has some new type hints included. This results in quite a few mypy issues that will need looking at. I'm setting this as high priority as I don't think we want to start pinning versions of commonly used libraries like xarray.

**Comment by rt17603** [comment]
The `mypy_updates` branch has addressed most of these.

The remaining one is as follows:
```
openghg/retrieve/met/_ecmwf.py:135: error: Argument 1 to "open_dataset" has incompatible type "bytes"; expected "Union[str, PathLike[Any], AbstractDataStore]"
```

**Comment by gareth-j** [comment]
This was fixed with #376 .
